app/services/action_reader.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+class ActionReader
+  attr_reader :query
+
+  DEFAULT_PER_PAGE = 4
+
+  def initialize(query)
+    @query = query
+  end
+
+  def run(page_number: 1, per_page: DEFAULT_PER_PAGE)
+    action_pagination = Action.where(query).page(page_number).per(per_page)
+    ActionCollator.run(action_pagination).push(action_pagination)
+  end
+
+  def csv(page_number: nil, per_page: DEFAULT_PER_PAGE)
```

Comment:
@NealJMD I really don't want the CSV download feature to be a part of Champaign. It's a blocking request, and a page with a large number of actions is going to punish Champaign's performance for our members (esp during traffic bursts). Could you move this to the worker (or even a lambda function! - happy to show you how) for async processing? You could save the CSV file to s3 and notify the campaigner by email, or even on slack!

Sorry for my comment. It's a lovely PR otherwise!